store/multiversion/memiterator.go

```diff
@@ -66,9 +69,47 @@ func (mi *memIterator) Value() []byte {
 		mi.abortChannel <- occtypes.NewEstimateAbort(val.Index())
 	}
 
+	// need to update readset
 	// if we have a deleted value, return nil
 	if val.IsDeleted() {
+		mi.ReadsetHandler.UpdateReadSet(key, nil)
```

Comment:
nit can this be a defer?

Reply:
yup, changed, although i'm not certain that it changes the outcome that much?